scripts/run-on-vm.sh

```
@@ -2,25 +2,48 @@
-# Save full run output under /tmp/ptp-operator (timestamped file; also shown on the terminal).
-mkdir -p /tmp/ptp-operator
-RUN_ON_VM_LOG="/tmp/ptp-operator/run-on-vm-$(date +%Y%m%d-%H%M%S).log"
+case "${RUN_ON_VM_VERBOSE}" in
+  1|true|TRUE|yes|YES|on|ON) RUN_ON_VM_VERBOSE=true ;;
+  *) RUN_ON_VM_VERBOSE=false ;;
+esac
+export RUN_ON_VM_VERBOSE
+
+# Per-run temp directory shared by all child scripts.
+REPO_ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd)"
+export PTP_RUN_DIR="${REPO_ROOT}/.local-runs/$(date +%Y%m%d-%H%M%S)"
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)